Add test set 1 - 4.

jwcullen opened the pull request:
Test set 1 covers: {LPCM, Opus} x {0, 1, 2, 3}-order ambisonics x {MONO, PROJECTED}.

Test set 2 covers: {LPCM, Opus} x {each layout except binaural}.

Test set 3 covers: {LPCM, Opus} x {1, 2, 3}-order ambisonics mixed with stereo.

Test set 4 covers: {LPCM, Opus} x {stereo, 3.1.2, 5.1} mixed with stereo.

jwcullen commented:
Updated with several test vectors in "Test Set 3" and "Test Set 4".

jingbo-marquis commented on tests/test_000060.textproto:
```diff
@@ -0,0 +1,150 @@
+    audio_elements {
+      audio_element_id: 300
+      mix_presentation_element_annotations {
+        audio_element_friendly_label: "test_sub_mix_0_audio_element_0"
```
count_label is 2, so the mix_presentation_element_annotations should have 2 audio_element_friendly_labels, refer [2.8. Mix Presentation OBU Syntax and Semantics](https://aomediacodec.github.io/iamf/#obu-mixpresentation)

jwcullen replied:
Resolved as of 31f5795.

jingbo-marquis reviewed (changes requested):
there are some problems in tc 60 and 231.

jwcullen commented:
Updated bitstream for test_000060 (with `count_label > 1`) to add a missing label .

jwcullen commented:
Added a simple test of bezier animation and fixed an enum value leading to an incorrect bitstream in test_000212.

jwcullen commented on tests/test_000231_decoded_substream_0.wav:
I deleted the file for now. Our library to write these files does not support 32-bit wav files. I opened an #44 because we'll have to get back this.

jwcullen commented:
Recent updates include the spec changes to `rendering_config` and including several rendered wav files. 

Please be aware some rendered wav files are omitted and will come in future commits.
Please be aware we have not updated to match the spec change introduced in AOMediaCodec/iamf#529 yet. So the bitstreams `num_subblocks` has not been updated with new behavior yet.

jwcullen commented:
Recent commits added some tests including explicitly provided recon gain, updated tagged sections based on a recent renumbering and all related reports, and fixed the change to `num_subblocks` in the bitstream.

I think we have captured the spec changes that broke the existing bitstreams. There are still more extension fields to test, but they are not always present in the bitstream.

yongmin-kwon reviewed (commented):
It seems to be another 'iamf' box in the 'iamf' box.
Please modify it after checking the specification.
(https://aomediacodec.github.io/iamf/#iasampleentry-section)

yongmin-kwon commented on tests/test_000071.textproto:
```diff
@@ -0,0 +1,531 @@
+  channel_ids: [0, 1]
+  channel_labels: ["L2", "R2"]
+}
+
```
For test_000071, there is no temporal_delimiter_metadata unlike other test vectors.
If it is not intended, please modify it.

jwcullen replied:
It was a mistake. Resolved in 632fc68

jwcullen commented:
> It seems to be another 'iamf' box in the 'iamf' box. Please modify it after checking the specification. (https://aomediacodec.github.io/iamf/#iasampleentry-section)

Ok try 326f8d7.

jwcullen commented:
Recent commits introduce some new test vectors to fill in coverage gaps and tag several sections that are broadly covered by many of the tests.

jwcullen commented:
Recent updates label more sections that were broad and covered by several test vectors already. In additional some test vectors with a binaural layout are added (with rendered wav files pending) and a set of AAC test vectors.

jwcullen commented:
Added binaural files. Please note when there is a binaural layout they will have the filename corresponding to that layout index (following the general pattern of most layouts). Otherwise they have a name based on the stereo layout index with "_binaural" appended (because there are 2 ways to interpret stereo layout).

jwcullen commented:
Recent updates fixed some issues with applying mix gain and some test vectors that were using duplicate parameter streams unintentionally. Also some missing files to compare rendering against are included and issues with some missing bytes in the Codec Config OBU for AAC were added.

yeroro commented:
I will merge this PR in preparation for the Spec review.
